packages/db-objects-redis/src/lib/objects/objectsInRedisClient.ts

```diff
@@ -2545,59 +2801,73 @@ class ObjectsInRedisClient {
+                    strObjs = await this.client.mget(keys);
                 } catch (e) {
                     return tools.maybeCallbackWithRedisError(callback, e);
                 }
 
                 const filteredKeys = [];
                 const filteredObjs = [];
-                objects = objects || [];
-                for (let k = 0; k < keys.length; k++) {
+
+                for (let i = 0; i < keys.length; i++) {
+                    const strObj = strObjs[i];
+                    let obj: null | ioBroker.Object;
                     try {
-                        objects[k] = JSON.parse(objects[k]);
+                        obj = strObj ? JSON.parse(strObj) : null;
                     } catch {
-                        this.log.error(`${this.namespace} Cannot parse JSON ${keys[k]}: ${objects[k]}`);
+                        this.log.error(`${this.namespace} Cannot parse JSON ${keys[i]}: ${strObj}`);
                         continue;
                     }
-                    if (!utils.checkObject(objects[k], options, CONSTS.ACCESS_WRITE)) {
+                    if (!utils.checkObject(obj, options, CONSTS.ACCESS_WRITE) || !obj) {
                         continue;
                     }
-                    if (!objects[k].acl) {
-                        objects[k].acl = {
+                    if (!obj.acl) {
+                        obj.acl = {
                             owner: (this.defaultNewAcl && this.defaultNewAcl.owner) || CONSTS.SYSTEM_ADMIN_USER,
                             ownerGroup:
                                 (this.defaultNewAcl && this.defaultNewAcl.ownerGroup) || CONSTS.SYSTEM_ADMIN_GROUP,
                             object:
                                 (this.defaultNewAcl && this.defaultNewAcl.object) ||
                                 CONSTS.ACCESS_USER_RW | CONSTS.ACCESS_GROUP_READ | CONSTS.ACCESS_EVERY_READ // '0644'
                         };
-                        if (objects[k].type === 'state') {
-                            objects[k].acl.state =
+                        if (obj.type === 'state') {
+                            // @ts-expect-error we have ensured, that property acl exists
+                            obj.acl.state =
                                 (this.defaultNewAcl && this.defaultNewAcl.state) ||
                                 CONSTS.ACCESS_USER_RW | CONSTS.ACCESS_GROUP_READ | CONSTS.ACCESS_EVERY_READ; // '0644'
                         }
                     }
                     if (options.object !== undefined) {
-                        objects[k].acl.object = options.object;
+                        obj.acl.object = options.object;
                     }
                     if (options.state !== undefined) {
-                        objects[k].acl.state = options.state;
+                        // @ts-expect-error TODO: state not allowed or types incorrect?
+                        obj.acl.state = options.state;
```

Comment:
AFAIK .state is only allowed on state objects. `obj` isn't guaranteed to be one here.